### PR TITLE
Clear UserReportingMetadataStaging records when users are deleted

### PR DESCRIPTION
**Pull request by sravfeyn**
On India, there are four old users for which `UserReportingMetadataStaging` records exist. Current code keeps spawning more tasks recursively if there are any records that exist. There is a condition where records were not deleted, so each task kept triggering more tasks. This aggravated after this [PR](https://github.com/dimagi/commcare-hq/pull/26305/files) on India specifically, because only on India the schedule is set to every 5 minutes (whereas on ICDS it's restricted to a certain time of the day) and has old records for deleted users.

This fix deletes the records that aren't to be processed and ~~stops spawning more subtasks~~

### CODE REVIEW

**Comment by sravfeyn** [comment]
@dannyroberts @snopoke

File: corehq/apps/users/tasks.py [hunk 1]
```diff
@@ -347,6 +348,3 @@ def process_reporting_metadata_staging():
                 user.save(fire_signals=False)
 
             record.delete()
-
-    if UserReportingMetadataStaging.objects.exists():
-        process_reporting_metadata_staging.delay()
```
Comment (snopoke) [inline corehq/apps/users/tasks.py]: without this it will only process 100 records

Comment (sravfeyn) [reply corehq/apps/users/tasks.py]: Ha, right. Reverted that change. Hopefully deleting the record when needed addresses this issue.

**Review by dannyroberts — APPROVED** [review]

**Comment by dannyroberts** [comment]
Looks like this was very successful: https://app.datadoghq.com/dashboard/bdu-k5b-bz5/celery-overview?from_ts=1579595715032&is_auto=false&live=false&page=0&tile_size=m&to_ts=1579618786000&tpl_var_celery_queue=%2A&tpl_var_celery_task=%2A&tpl_var_environment=india&fullscreen_widget=6682969619012402&fullscreen_section=overview. Thanks for the quick and effective response!

**Pull request merged**